compiler/rustc_mir_build/src/build/matches/test.rs

```diff
@@ -318,33 +318,41 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             }
 
             TestKind::Range(ref range) => {
-                let lower_bound_success = self.cfg.start_new_block();
-
+                let [success, fail] = *target_blocks else {
+                    bug!("`TestKind::Range` should have two target blocks");
+                };
                 // Test `val` by computing `lo <= val && val <= hi`, using primitive comparisons.
-                // FIXME: skip useless comparison when the range is half-open.
-                let lo = range.lo.to_const(range.ty, self.tcx);
-                let hi = range.hi.to_const(range.ty, self.tcx);
-                let lo = self.literal_operand(test.span, lo);
-                let hi = self.literal_operand(test.span, hi);
                 let val = Operand::Copy(place);
 
-                let [success, fail] = *target_blocks else {
-                    bug!("`TestKind::Range` should have two target blocks");
+                let intermediate_block = if !range.lo.is_finite() {
+                    block
```

Comment:
I'm a little surprised that you can handle both positive and negative infinity the same way in  the code here (`range.lo`) and immediate below (`range.hi`). 

What am I missing here? Is there some assumption/guarantee that you will never see a positive infinity on lo nor a negative infinity on hi?

Reply:
Absolutely yep. I think this is documented in PatRange but if not it should be

Reply:
It wasn't documented so I added a note

Reply:
I find assertions superior to comments, but I'll live with this. It doesn't help that the struct in question is a bunch of pub fields, so we cannot enforce this invariant locally (e.g. by asserting just in the constructors)